slothy/core/slothy.py

```diff
@@ -464,7 +463,6 @@ def loop_lbl_iter(i):
             for i in range(1, num_exceptional):
                 optimized_code += indented(self.arch.Branch.if_equal(loop_cnt, i, loop_lbl_iter(i)))
 
-        loop = self.arch.Loop(lbl_start=loop_lbl)
```

Comment:
Was this just. dead?

Reply:
No, but `loop` is now initialized as part of the extraction earlier ([line 400](https://github.com/slothy-optimizer/slothy/blob/f5fba205e985e2acf82b8043b57530abf334fff6/slothy/core/slothy.py#L400)) because the type of loop needs to be determined for the `start` method to be appropriate in the followig.

Reply:
Got it, thank you